src/modules/ekf2/EKF2.cpp

```diff
@@ -1741,8 +1740,7 @@ int EKF2::task_spawn(int argc, char *argv[])
 		bool ekf2_instance_created[4][4] {}; // IMUs * mags
 
 		while ((multi_instances_allocated < multi_instances)
-		       && (vehicle_status_sub.get().arming_state != vehicle_status_s::ARMING_STATE_ARMED)
-		       && (hrt_elapsed_time(&time_started) < 30_s)) {
```

Comment:
This is running from the `ekf2` frontend, which means you can't use `ekf2 status` until it exits. 

What about bumping the time to 60 or even 120 s? This is also a problem for UAVCAN nodes that can take quite a while to start if the firmware is being updated.

Reply:
If it ever stops it will mean that HITL doesn't work.

Reply:
How about not exiting early if no instances have started successfully (multi_instances_allocated = 0), otherwise run for up to 120s or until armed? I think that would cover most real usage and keep the frontend available for interactive bench testing.

Reply:
Or we make it dependent on hil mode or not.

Reply:
@dagar I'm not sure what you mean actually. `ekf2 status` works just fine with this change:
```
nsh> ekf2 status

ekf2:0 attitude: 0, local position: 0, global position: 0
ekf2: ECL update: 0 events, 0us elapsed, 0.00us avg, min 0us max 0us 0.000us rms
ekf2: ECL full update: 0 events, 0us elapsed, 0.00us avg, min 0us max 0us 0.000us rms
ekf2: IMU message missed: 0 events
```

Reply:
With multi-ekf enabled this will continue running until it finds all the sensors specified by EKF2_MULTI_IMU and EKF2_MULTI_MAG. On a real system this might keep running for a bit waiting for a UAVCAN mag, etc.

Reply:
I've changed it so now ekf2 only tries forever in HITL mode.